### PR TITLE
Added inline commenting 

### PR DESCRIPTION
**Pull request by harshdoesdev**
Fixed the remove_inline_comments function:
- Fixed the escape sequence bug which was preventing the `;` from being escaped when using `\\` double-backslashes
- Changed the regular while loop with while..some while loop
- Fixed formatting issues

### CODE REVIEW

File: ftd/src/ftd2021/p1/parser.rs [hunk 1]
```diff
@@ -401,6 +408,71 @@ pub fn parse_file_for_global_ids(data: &str) -> Vec<(String, usize)> {
...
+        };
+        ($s:expr, $t: expr) => {
+            assert_eq!(super::remove_inline_comments($s), $t)
+        };
+    }
+
+    #[test]
+    fn test_esc() {
+        t!(r#"Hello      ;; displays the hello"#, "Hello");
+        t!(r#"Hello      \;; displays the hello"#, "Hello      ;");
```
Comment (amitu) [inline ftd/src/ftd2021/p1/parser.rs:471]: This is wrong. There is no comment in this line as comment requires two consecutive unescaped `;;`.

**Comment by harshdoesdev** [comment]
More updates about why some tests are failing:
https://github.com/fastn-stack/fastn/issues/958#issuecomment-1641429175

**Comment by amitu** [comment]
> Currently, we have used ;; in many places as a marker for highlighting HTML (Refer: HTML test 52) in FTD, which is why some tests are failing. 

We have to change the syntax for code highlighting.

**Comment by amitu** [comment]
For code highlighting we have to put `# <hl>` etc. We will honour line comment based on lang attribute. For each lang lets create a map for line comment. Or we just assume that virtually everyone either uses `#` or `//` as line comment, so we look for `# <hl>` and `// <hl>`, in any other language, people can use either of the two.

File: ftd/src/p1/parser.rs [hunk 2]
```diff
@@ -872,9 +872,70 @@ fn clean_line(line: &str) -> String {
     if line.starts_with("\\;;") || line.starts_with("\\-- ") {
         return line[1..].to_string();
     }
+
+    if !line.contains(" ;; <hl>") {
```
Comment (Heulitig) [inline ftd/src/p1/parser.rs:876]: Just change it to this

```suggestion
    // Ignoring <hl> to ignore code highlighting comments
    // todo: need to fix the code highlighting syntax
    if !line.contains("<hl>") {
```

File: ftd/src/p1/parser.rs [hunk 3]
```diff
@@ -872,9 +872,70 @@ fn clean_line(line: &str) -> String {
...
     line.to_string()
 }
 
+fn remove_inline_comments(line: &str) -> String {
+    let mut output = String::new();
+
+    let mut chars = line.chars().peekable();
+
+    let mut escape = false;
+
```
Comment (Heulitig) [inline ftd/src/p1/parser.rs:889]:
```suggestion
    let mut output = String::new();
    let mut chars = line.chars().peekable();
    let mut escape = false;
    let mut count = 0;
```

**Pull request merged**